packages/server/src/resources/course.rs

```diff
@@ -28,6 +28,27 @@ pub enum Tag {
     English,
     Malag,
     Sport,
+    SportTeam, // TODO: check if need this
+}
+
+impl Course {
+    pub fn is_english(&self) -> bool {
+        // TODO: change it to "is_some_and()" when become stable
+        self.tags
+            .clone()
+            .unwrap_or_default()
+            .contains(&Tag::English)
+    }
+
+    pub fn is_sport(&self) -> bool {
+        // TODO: change it to "is_some_and()" when become stable
+        self.tags.clone().unwrap_or_default().contains(&Tag::Sport)
+    }
+
+    pub fn is_malag(&self) -> bool {
+        // TODO: change it to "is_some_and()" when become stable
+        self.tags.clone().unwrap_or_default().contains(&Tag::Malag)
+    }
```

Comment:
I thought of a cool way to generalize this to avoid the small code duplication:
```rust
    fn is(&self, tag: Tag) -> bool {
        // TODO: change it to "is_some_and()" when become stable
        self.tags.clone().unwrap_or_default().contains(&tag)
    }
    pub fn is_english(&self) -> bool {
        self.is(Tag::English)
    }

    pub fn is_sport(&self) -> bool {
        self.is(Tag::Sport)
    }

    pub fn is_malag(&self) -> bool {
        self.is(Tag::Malag)
    }
```
Notice that `fn is(&self) -> bool` is private on purpose.
WDYT?

Reply:
I will take it.